tests/rule-grouping/rule-grouping-4/test.yaml

```diff
@@ -0,0 +1,151 @@
+requires:
+  min-version: 8
+
+pcap: false
+
+args:
+  - --engine-analysis
+
+checks:
+  - filter:
+      filename: rule_group.json
+      count: 1
+      match:
+        tcp.toserver[0].port: 7
+        tcp.toserver[0].port2: 7
+        tcp.toserver[0].rulegroup.id: 0
+        tcp.toserver[0].rulegroup.rules[0].sig_id: 1
+        tcp.toserver[0].rulegroup.rules[1].sig_id: 2
+        tcp.toserver[0].rulegroup.rules[2].sig_id: 3
+        tcp.toserver[0].rulegroup.rules[3].sig_id: 6
+  - filter:
```

Comment:
merge into 1

Reply:
This is on TODO (ref: https://github.com/OISF/suricata/pull/10470/files#diff-26baf730c742afac833c40d8c1927c2040dcf619b387647e60f2aa34b4a9d58dR1168) Working on this today. Turns out I don't need to wait for the later stage, I can do this right after having found overlaps.